src/log.rs

```diff
@@ -64,14 +63,52 @@ macro_rules! emit_event {
 pub trait LogExt<T> {
     /// Emits a warning if the receiver contains an Err value.
     ///
+    /// If you want to add context to the log, use `log_m()`.
+    ///
     /// Returns an [`Option<T>`] with the `Ok(_)` value, if any:
```

Comment:
WRT naming: See https://github.com/deltachat/deltachat-core-rust/pull/2255#discussion_r584890889 for some previous discussion. But I feel that we should decide this one last as it slightly depends on some other things and is the least important decision, I feel.

WRT explicit `.ok()`: Usually after logging the error, we don't need the error anymore; when going through `lib.rs` I couldn't find a single case where `log_err()` was used and then the error value was still used.

> It makes sense for logging calls to not implicitly do something else like changing the return type

To me, it makes totally sense :) The Err case is taken care of now, so that we only need the Ok value anymore now.